### PR TITLE
Master pos investigate random error jcb

### PR DESCRIPTION
**Pull request by pimodoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by rco-odoo** [comment]
@pimodoo

File: odoo/addons/base/models/ir_property.py [hunk 1]
```diff
@@ -167,6 +175,34 @@ def get_by_record(self):
...
+        :param model: the field's model name
+        :param value: the field's value
+        :param company: the company (record or id)
+        """
+        field_id = self.env['ir.model.fields']._get(model, name).id
+        company_id = bool(company) and int(company)
+        prop = self.search([
+            ('fields_id', '=', field_id),
+            ('company_id', '=', company_id),
+            ('res_id', '=', False),
```
[the remark this answers is not in the document]
Comment (rco-odoo) [reply odoo/addons/base/models/ir_property.py]: Nope: to set the field `foo` on `record` in `company`, the *right way* is to use
```python
record.with_company(company).foo = value
```

[the remark this answers is not in the document]
Comment (xmo-odoo) [reply odoo/addons/base/models/ir_property.py]: I still agree with @pimodoo that the method should be called `set_default` if the intent is to set the global or per-company default. Even more so as one would expect `get` and `set` to match, which is very much not the case here.

**Comment by rco-odoo** [comment]
@xmo-odoo @pimodoo changed the name of the method to `set_default`.
That's indeed clearer.

**Comment by xmo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**